src/sulogin-shell/sulogin-shell.c

```diff
@@ -103,17 +106,33 @@ int main(int argc, char *argv[]) {
                 /* allows passwordless logins if root account is locked. */
                 sulogin_cmdline[1] = "--force";
 
-        (void) fork_wait(sulogin_cmdline);
-
         r = bus_connect_system_systemd(&bus);
         if (r < 0) {
                 log_warning_errno(r, "Failed to get D-Bus connection: %m");
                 r = 0;
-        } else {
+        }
+
+        path = unit_dbus_path_from_name(SPECIAL_DEFAULT_TARGET);
+        if (!path)
+                return log_oom();
```

Comment:
I see. I'll do it.

Reply:
Wait. It might be here correct to discard D-Bus related error status just like done in fork_wait() and reload_manager(). systemd-sulogin should keep at least sulogin in the abnormal situation where D-Bus doesn't work well. Then, DEFINE_MAIN_FUNCTION() is not needed because there's no error handling in main(). How do you think?

Reply:
I am not sure I correctly understand your point, but we usually use `DEFINE_MAIN_FUNCTION()` even the original main function always succeeds. So, when you touch the main, then please convert to use the macro.